workflow/controller/operator.go

```diff
@@ -825,6 +825,11 @@ func (woc *wfOperationCtx) podReconciliation() error {
 			woc.wf.Status.Nodes[nodeID] = node
 			woc.log.Warnf("pod %s deleted", nodeID)
 			woc.updated = true
+		} else {
+			// At this point we are certain that the pod associated with our node is running or has been run;
+			// it is safe to extract the k8s-node information given this knowledge.
+			node.HostNodeName = seenPods[nodeID].Spec.NodeName
+			woc.wf.Status.Nodes[nodeID] = node
```

Comment:
not sure whether or not it is necessary to put `woc.updated = true` here?

Reply:
It is necessary.

The way the controller works is by loading the `Workflow` CRD from K8s into memory, operating on it, and rewriting all the updates to the CRD if necessary. `woc.updated` is the flag that tells the controller that there are updates to be rewritten to the CRD: in this case saving the `HostNodeName` to the `NodeStatus`.